### PR TITLE
Support None gradient in split_axis

### LINKED ISSUE #308: split_axis.backward fails on incomplete gradients
When there is a None in the grad_outputs, split_axis fails to backprop the incomplete gradients.

### PR DESCRIPTION
**Pull request by unnonouno**
fix #308

### CODE REVIEW

File: chainer/functions/split_axis.py [hunk 1]
```diff
@@ -85,14 +86,25 @@ def forward_gpu(self, x):
         return tuple(ys)
 
     def backward_cpu(self, x, gys):
-        return numpy.concatenate(gys, axis=self.axis),
+        if any(gy is None for gy in gys):
```
Comment (beam2d) [inline chainer/functions/split_axis.py]: Is this _if_ statement still needed?

Comment (unnonouno) [inline chainer/functions/split_axis.py]: No, but i don't know efficiency.

File: tests/functions_tests/test_split_axis.py [hunk 2]
```diff
@@ -93,4 +93,31 @@ def setUp(self):
         self.axis = 0
 
 
+class TestSplitAxisNone(unittest.TestCase):
+
+    def setUp(self):
+        self.x = numpy.array([1, 2], dtype=numpy.float32)
+        self.ys = [self.x[:1], self.x[1:]]
```
Comment (beam2d) [inline tests/functions_tests/test_split_axis.py]: Seems unused?

**Comment by beam2d** [comment]
LSTM except above comments!

File: chainer/functions/split_axis.py (hunk 1 above)
Comment (beam2d) [inline chainer/functions/split_axis.py]: OK, keep as is for efficiency

**Comment by beam2d** [comment]
[LGTM](http://lgtm.in/i/86LEpXXUp)
LGTM!

**Pull request merged**